src/wristpy/processing/metrics.py

```diff
@@ -53,3 +54,192 @@ def angle_relative_to_horizontal(
+    acceleration_grouped_by_short_window = _group_acceleration_data_by_time(
+        acceleration, short_epoch_length
+    )
+
+    nonwear_value_array = _compute_nonwear_value_array(
+        acceleration_grouped_by_short_window,
+        n_short_epoch_in_long_epoch,
+        std_criteria,
+        range_criteria,
+    )
+
+    nonwear_value_array_cleaned = _cleanup_isolated_ones_nonwear_value(
+        nonwear_value_array
+    )
+    non_wear_flag = np.where(nonwear_value_array_cleaned >= 2, 1, 0)
+
+    return models.Measurement(
+        measurements=non_wear_flag, time=acceleration_grouped_by_short_window["time"]
+    )
+
+
+def _group_acceleration_data_by_time(
+    acceleration: models.Measurement, window_length: int
+) -> pl.DataFrame:
+    """Helper function to group the acceleration data by short windows.
+
+    Args:
+        acceleration: The Measurment instance that contains the calibrated acceleration.
+        window_length: The window size, in seconds.
+
+    Returns:
+        A polars DataFrame with the acceleration data grouped by window_length.
+    """
+    acceleration_polars_df = pl.DataFrame(
+        {
+            "X": acceleration.measurements[:, 0],
+            "Y": acceleration.measurements[:, 1],
+            "Z": acceleration.measurements[:, 2],
+            "time": acceleration.time,
+        }
+    )
+    acceleration_polars_df = acceleration_polars_df.with_columns(
+        pl.col("time").set_sorted()
+    )
+
+    acceleration_grouped_by_window_length = acceleration_polars_df.group_by_dynamic(
+        index_column="time", every=(str(window_length) + "s")
+    ).agg([pl.all().exclude(["time"])])
+
+    return acceleration_grouped_by_window_length
+
+
+def _compute_nonwear_value_array(
+    grouped_acceleration: pl.DataFrame,
+    n_short_epoch_in_long_epoch: int,
+    std_criteria: float,
+    range_criteria: float,
+) -> np.ndarray:
+    """Helper function to calculate the nonwear value array.
+
+    This function calculates the nonwear value array based on the GGIR 2023 methodology.
+    It computes the nonwear value for each axis, based on the acceleration data that
+    makes up one long epoch window. That nonwear value is then applied to all the
+    short windows that make up the long window. It iterates forward by one short_window
+    length and repeats the process. For the overlapping short windows, the maximum
+    nonwear value is kept and is assigned to the nonwear value array.
+
+    Args:
+        grouped_acceleration: The acceleration data grouped into short windows.
+        n_short_epoch_in_long_epoch: Number of short epochs that makeup one long epoch.
+        std_criteria: Threshold criteria for standard deviation.
+        range_criteria: Threshold criteria for range of acceleration.
+
+    Returns:
+        Non-wear value array.
+    """
+    nonwear_value_array = np.zeros(len(grouped_acceleration))
+
+    for window_n in range(len(grouped_acceleration) - n_short_epoch_in_long_epoch + 1):
```

Comment:
Can `len(...)` be puleld out of this statement and given a proper name? What are we looping over? `n_windows`?

Reply:
Yes, it is the total number of short windows. Will pull it out.